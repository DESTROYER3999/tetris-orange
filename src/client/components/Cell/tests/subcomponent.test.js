import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import CellSub from "../subcomponent";

describe("<CellSub />", () => {
  test("Cell rendering without color, letter and is not Spectre", () => {
    const wrapper = render(<CellSub color="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering with color, without letter and is not spectre", () => {
    const wrapper = render(<CellSub color="#000000" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering with letter, without color and is not spectre", () => {
    const wrapper = render(<CellSub color="" letter="P" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering with letter and color, is not spectre", () => {
    const wrapper = render(<CellSub color="#000000" letter="P" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering as spectre, without color and letter", () => {
    const wrapper = render(<CellSub color="" isSpectre={true} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering as spectre, with color and without letter", () => {
    const wrapper = render(<CellSub color="#000000" isSpectre={true} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering as spectre, with letter and without color", () => {
    const wrapper = render(<CellSub isSpectre={true} letter="P" color="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Cell rendering as spectre, with letter and color", () => {
    const wrapper = render(
      <CellSub isSpectre={true} letter="P" color="#000000" />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
