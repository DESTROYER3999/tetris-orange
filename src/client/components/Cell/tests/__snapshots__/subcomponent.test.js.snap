// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellSub /> Cell rendering as spectre, with color and without letter 1`] = `
<div
  class="sc-bdVaJa mieYS"
  color="#000000"
>
  <p
    class="sc-bwzfXH dfRfSZ"
  />
</div>
`;

exports[`<CellSub /> Cell rendering as spectre, with letter and color 1`] = `
<div
  class="sc-bdVaJa mieYS"
  color="#000000"
>
  <p
    class="sc-bwzfXH dfRfSZ"
  >
    P
  </p>
</div>
`;

exports[`<CellSub /> Cell rendering as spectre, with letter and without color 1`] = `
<div
  class="sc-bdVaJa bPijFL"
  color=""
>
  <p
    class="sc-bwzfXH dfRfSZ"
  >
    P
  </p>
</div>
`;

exports[`<CellSub /> Cell rendering as spectre, without color and letter 1`] = `
<div
  class="sc-bdVaJa bPijFL"
  color=""
>
  <p
    class="sc-bwzfXH dfRfSZ"
  />
</div>
`;

exports[`<CellSub /> Cell rendering with color, without letter and is not spectre 1`] = `
<div
  class="sc-bdVaJa ejDKVY"
  color="#000000"
>
  <p
    class="sc-bwzfXH dfRfSZ"
  />
</div>
`;

exports[`<CellSub /> Cell rendering with letter and color, is not spectre 1`] = `
<div
  class="sc-bdVaJa ejDKVY"
  color="#000000"
>
  <p
    class="sc-bwzfXH dfRfSZ"
  >
    P
  </p>
</div>
`;

exports[`<CellSub /> Cell rendering with letter, without color and is not spectre 1`] = `
<div
  class="sc-bdVaJa idqHKJ"
  color=""
>
  <p
    class="sc-bwzfXH dfRfSZ"
  >
    P
  </p>
</div>
`;

exports[`<CellSub /> Cell rendering without color, letter and is not Spectre 1`] = `
<div
  class="sc-bdVaJa idqHKJ"
  color=""
>
  <p
    class="sc-bwzfXH dfRfSZ"
  />
</div>
`;
